lib/mediakit_player.dart

```diff
@@ -158,60 +211,76 @@ class MediaKitPlayer extends AudioPlayerPlatform {
       _dataController.stream;
 
   /// Updates the playback event
-  void _updatePlaybackEvent(
-      {Duration? duration, IcyMetadataMessage? icyMetadata}) {
+  void _updatePlaybackEvent() {
     _eventController.add(PlaybackEventMessage(
       processingState: _processingState,
       updateTime: DateTime.now(),
       updatePosition: _position,
       bufferedPosition: _bufferedPosition,
-      duration: duration,
-      icyMetadata: icyMetadata,
+      duration: _duration,
+      icyMetadata: null,
       currentIndex: _currentIndex,
       androidAudioSessionId: null,
+      errorCode: _errorCode,
+      errorMessage: _errorMessage,
     ));
   }
 
   @override
   Future<LoadResponse> load(LoadRequest request) async {
     _logger.finest('load(${request.toMap()})');
+    _mediaOpened = false;
+    _loadCompleter = Completer();
     _currentIndex = request.initialIndex ?? 0;
     _bufferedPosition = Duration.zero;
     _position = Duration.zero;
-
-    _processingState = ProcessingStateMessage.buffering;
+    _duration = null;
+    _processingState = ProcessingStateMessage.loading;
```

Comment:
maybe this should switch between loading and idle (based on the _playing status)?